### PR TITLE
Initialize state with mt19937 random numbers

### PR DESCRIPTION
**Pull request by christopherwharrop-noaa**
Add mt19937 routines for computing random numbers so that initialization is reproducible across compilers/platforms.

Add routines for printing state variable statistics for verification.

### CODE REVIEW

**Review by dshawul — APPROVED** [review]

**Review by dshawul — CHANGES_REQUESTED** [review]
--

**Review by danielabdi-noaa — APPROVED** [review]
This is a pretty nice way of dealing with inconsistency in random number generation across compilers. I have tested the CPU version and like the way all variable state is printed before and after computation. This should enable ctest easily.

**Comment by robgpita** [comment]
Looks good! Looking forward to having the input data included in this repo to run the test suite.

File: ref/src/kernel.F90 [hunk 1]
```diff
@@ -341,6 +344,202 @@ SUBROUTINE CRTM_Doubling_layer_AD(n_streams, & ! Input, number of streams
...
+  !------------------------------------------------------------------
+  SUBROUTINE print_2d_variable(name, data)
+
+    character(len=*) :: name
+    real(fp)         :: data(:,:)
+
+    ! Note: Assumed shape array sections always start with index=1 for all
+    ! dimensions
+    !       So we don't have to know start/end indices here
+    WRITE(*,'(A5, A17,5ES20.10)') "TEST ", name, minval(data), maxval(data), data(1,1), &
```
Comment (robgpita) [inline ref/src/kernel.F90:481]: Glad the lessons learned from c_sw to print scientific notation in compliance with C standards made it in here.

**Pull request merged**